File: apps/server/internal/application/dto/receipt.go
package dto

// SendReceiptRequest represents a request to send read receipts for messages
type SendReceiptRequest struct {
	SessionID  string   `json:"session_id" validate:"required,uuid"`
	ChatJID    string   `json:"chat_jid" validate:"required"`
	MessageIDs []string `json:"message_ids" validate:"required,min=1,dive,required"`
}

// ReceiptResponse represents the response after sending receipts
type ReceiptResponse struct {
	ProcessedCount int    `json:"processed_count"`
	Timestamp      string `json:"timestamp"`
}

// NewReceiptResponse creates a ReceiptResponse
func NewReceiptResponse(count int, timestamp string) *ReceiptResponse {
	return &ReceiptResponse{
		ProcessedCount: count,
		Timestamp:      timestamp,
	}
}

File: packages/schema/src/receipt.ts
/**
 * Receipt-related schemas
 * Matches backend: dto/receipt.go
 */
import { z } from "zod";
import { jidSchema, sessionIdSchema, timestampSchema } from "./common";

/**
 * Send receipt request schema
 * Matches: dto.SendReceiptRequest
 */
export const sendReceiptRequestSchema = z.object({
  session_id: sessionIdSchema,
  chat_jid: jidSchema,
  message_ids: z.array(z.string().min(1)).min(1),
});

export type SendReceiptRequest = z.infer<typeof sendReceiptRequestSchema>;

/**
 * Receipt response schema
 * Matches: dto.ReceiptResponse
 */
export const receiptResponseSchema = z.object({
  processed_count: z.number().int().min(0),
  timestamp: timestampSchema,
});

export type ReceiptResponse = z.infer<typeof receiptResponseSchema>;
